src/Common/ThreadFuzzer.cpp

```diff
@@ -361,7 +361,7 @@ void ThreadFuzzer::setup() const
 
 /// Starting from glibc 2.34 there are no internal symbols without version,
 /// so not __pthread_mutex_lock but __pthread_mutex_lock@2.2.5
-#if defined(OS_LINUX) and !defined(USE_MUSL)
+#if defined(OS_LINUX) and !defined(USE_MUSL) and !defined(__loongarch64)
```

Comment:
It is not really clear to me why the special handling for loongarch64 is needed here (l. 364, l. 387). Can't we define a normal branch instead (like l. 374)?

Reply:
In glibc>=2.36, '__pthread_mutex_lock' and '__pthread_mutex_unlock' do not exist:

```sh
# s390's pthread_mutex_lock symbol table
# readelf -sW contrib/sysroot/linux-s390x/s390x-linux-gnu/libc/lib64/libpthread.so.0 | grep  "pthread_mutex_lock"
   168: 000000000000a0c0   760 FUNC    WEAK   DEFAULT   13 pthread_mutex_lock@@GLIBC_2.2
   199: 000000000000a0c0   760 FUNC    GLOBAL DEFAULT   13 __pthread_mutex_lock@@GLIBC_2.2
    39: 0000000000009b58  1380 FUNC    LOCAL  DEFAULT   13 __pthread_mutex_lock_full
   195: 000000000000a0c0   760 FUNC    LOCAL  DEFAULT   13 __GI___pthread_mutex_lock
   436: 000000000000a0c0   760 FUNC    WEAK   DEFAULT   13 pthread_mutex_lock
   563: 000000000000a0c0   760 FUNC    GLOBAL DEFAULT   13 __pthread_mutex_lock

# loongarch64's pthread_mutex_lock symbol table
# readelf -sW contrib/sysroot/linux-loongarch64/lib/loongarch64-linux-gnu/libc.so.6 | grep  "pthread_mutex_lock"
  1265: 0000000000081578   828 FUNC    GLOBAL DEFAULT   13 pthread_mutex_lock@@GLIBC_2.36
```

The minimum version of glibc for 'loongarch64' is 2.36, so we can't define a normal branch instead (like l. 374).